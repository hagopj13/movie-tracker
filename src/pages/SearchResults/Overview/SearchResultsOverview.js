import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { makeStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';

import searchSelectors from 'store/search/search.selectors';

type Props = {
  searchQuery: string,
  totalResults: number,
};

const useStyles = makeStyles((theme) => ({
  root: {
    fontWeight: theme.typography.fontWeightRegular,
  },
  bold: {
    fontWeight: theme.typography.fontWeightBold,
  },
}));

export const SearchResultsOverview = (props: Props) => {
  const { searchQuery, totalResults } = props;

  const classes = useStyles();

  return (
    <Typography className={classes.root} variant="h6">
      Found
      <span className={classes.bold}>{` ${totalResults} `}</span>
      results for
      <span className={classes.bold}>{` ${searchQuery} `}</span>
    </Typography>
  );
};

const mapStateToProps = createStructuredSelector({
  searchQuery: searchSelectors.selectQuery,
  totalResults: searchSelectors.selectTotalResults,
});

export default connect(mapStateToProps)(SearchResultsOverview);
