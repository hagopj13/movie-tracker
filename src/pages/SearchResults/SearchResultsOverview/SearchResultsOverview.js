import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { makeStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';

import searchSelectors from 'store/search/search.selectors';

type Props = {
  searchQuery: string,
  totalResults: number,
};

const useStyles = makeStyles((theme) => ({
  root: {
    fontWeight: theme.typography.fontWeightRegular,
  },
  bold: {
    display: 'inline',
    fontWeight: theme.typography.fontWeightBold,
  },
}));

const SearchResultsOverview = (props: Props) => {
  const { searchQuery, totalResults } = props;

  const classes = useStyles();

  return (
    <Typography className={classes.root} variant="h6">
      Found
      <Typography className={classes.bold} variant="h6">
        {` ${totalResults} `}
      </Typography>
      results for
      <Typography className={classes.bold} variant="h6">
        {` ${searchQuery} `}
      </Typography>
    </Typography>
  );
};

const mapStateToProps = createStructuredSelector({
  searchQuery: searchSelectors.selectQuery,
  totalResults: searchSelectors.selectTotalResults,
});

export default connect(mapStateToProps)(SearchResultsOverview);
