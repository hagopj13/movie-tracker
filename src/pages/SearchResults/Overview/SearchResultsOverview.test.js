import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';

import { SearchResultsOverview } from './SearchResultsOverview';

describe('SearchResultsOverview component', () => {
  let shallow;
  let wrapper;

  beforeAll(() => {
    shallow = createShallow();
  });

  beforeEach(() => {
    const mockProps = {
      searchQuery: 'Some search query',
      totalResults: 100,
    };
    wrapper = shallow(<SearchResultsOverview {...mockProps} />);
  });

  it('should render the SearchResultsOverview component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
