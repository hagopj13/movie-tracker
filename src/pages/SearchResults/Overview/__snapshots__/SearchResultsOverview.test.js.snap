// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResultsOverview component should render the SearchResultsOverview component correctly 1`] = `
<WithStyles(ForwardRef(Typography))
  className="makeStyles-root-1"
  variant="h6"
>
  Found
  <span
    className="makeStyles-bold-2"
  >
     100 
  </span>
  results for
  <span
    className="makeStyles-bold-2"
  >
     Some search query 
  </span>
</WithStyles(ForwardRef(Typography))>
`;
